aips/on-chain-randomness-impl.md

```diff
@@ -0,0 +1,419 @@
+---
+aip: (this is determined by the AIP Manager, leave it empty when drafting)
+title: Implementation of instant on-chain randomness
+author: Alin Tomescu (alin@aptoslabs.com), Zhuolun "Daniel" Xiang (daniel@aptoslabs.com), Zhoujun Ma (zhoujun@aptoslabs.com)
+discussions-to (*optional): <a url pointing to the official discussion thread>
+Status: Draft
+last-call-end-date (*optional): <mm/dd/yyyy the last date to leave feedbacks and reviews>
+type: Standard (Core, Networking, Framework)
+created: 02/12/2024
+updated (*optional): <mm/dd/yyyy>
+requires (*optional): <AIP number(s)>
+---
+
+# AIP-X - Implementation of instant on-chain randomness
+
+## Summary
+
+This AIP describes the implementation of the randomness API from AIP-41[^aip-41], focusing on the cryptographic constructions and their efficient integration into our system. Specifically, we describe how we use a weighted distributed key generation (DKG) protocol in a PoS setting to set up threshold keys among the validators, and have validators generate randomness every block using a weighted verifiable unpredictable function (VUF). We will also describe other required changes such as reconfiguration change.
```

Comment:
this summary currently describes how we are doing the implementation. can we add a few lines about the goals and motivation behind this aip in the summary?

Reply:
done